api/src/main/java/org/pmiops/workbench/monitoring/StackdriverStatsExporterService.java

```diff
@@ -25,93 +23,41 @@
   private static final Logger logger =
       Logger.getLogger(StackdriverStatsExporterService.class.getName());
   private static final String STACKDRIVER_CUSTOM_METRICS_PREFIX = "custom.googleapis.com/";
-  private static final String MONITORED_RESOURCE_TYPE = "generic_node";
-  private static final String PROJECT_ID_LABEL = "project_id";
-  private static final String LOCATION_LABEL = "location";
-  private static final String NAMESPACE_LABEL = "namespace";
-  private static final String NODE_ID_LABEL = "node_id";
-  public static final String UNKNOWN_INSTANCE_PREFIX = "unknown-";
 
-  private boolean initialized;
   private Provider<WorkbenchConfig> workbenchConfigProvider;
-  private ModulesService modulesService;
+  private Provider<MonitoredResource> monitoredResourceProvider;
 
   public StackdriverStatsExporterService(
-      Provider<WorkbenchConfig> workbenchConfigProvider, ModulesService modulesService) {
+      @Qualifier(CacheSpringConfiguration.WORKBENCH_CONFIG_SINGLETON)
+          Provider<WorkbenchConfig> workbenchConfigProvider,
+      Provider<MonitoredResource> monitoredResourceProvider) {
```

Comment:
Since monitoredResourceProvider is created using Singleton workbenchConfigProvider above in MonitoringSpringConfiguration, why do we still need a reference to workbenchConfigProvider here?

Reply:
+1, I don't think there are any situations where we'd want to inject the SINGULAR qualified provider - only rationale I could see would be for testing purposes

Reply:
Very good eye. We don't, really.